### PR TITLE
fix proxy log search

### LINKED ISSUE #6829: DiagnosticsService can not filter filename by datatime
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

1. make a cluster up
2. add a new log file like 'tiflash_tikv-2023-01-30T02-00-12.039.log' in tiflash/log
3. do log search in dashboard with time selection

### 2. What did you expect to see? (Required)

we read the log match the time selection.

### 3. What did you see instead (Required)

we read all log files in tiflash/log(when there are quite a lot of log files , search will be very slow)
image

This problem is related to the code FilterFileByDatetime in LogSearch.cpp, the filter function is not appropriate with the log file's name now.

### 4. What is your TiFlash version? (Required)
nightly

### PR DESCRIPTION
**Pull request by hehechen**
### What problem does this PR solve?

Issue Number: close #6829

Problem Summary:
Fix proxy log search since the format of TiKV log filename has been modified in https://github.com/tikv/tikv/pull/11657
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by solotzg — APPROVED** [review]
lgtm

**Comment by JaySon-Huang** [comment]
I remember once that because of the log filename format, tiup clinic can filter the log file when collecting within a specified time range.
@hehechen Can you also ensure whether that problem still exists? Should the tiflash log filename format align with tikv's newest one?

File: dbms/src/Flash/tests/gtest_log_search.cpp [hunk 1]
```diff
@@ -123,7 +123,7 @@ TEST_F(LogSearchTest, SearchDir)
     ASSERT_TRUE(FilterFileByDatetime("/1/server.log.2021-10-09-14:50:55.481.gz", {"/1/test-err.log"}, 1633855377000)); // 1633855377000 : 2021-10-10 16:42:57
     ASSERT_FALSE(FilterFileByDatetime("/1/server.log.2021-10-10-16:43:57.123.gz", {"/1/test-err.log"}, 1633855377000));
 
-    ASSERT_TRUE(FilterFileByDatetime("/1/proxy.log.2021-10-09-14:50:55.123456789", {"/1/test-err.log"}, 1633855377000));
+    ASSERT_TRUE(FilterFileByDatetime("/1/proxy.log.2021-10-09T14:50:55.123.log", {"/1/test-err.log"}, 1633855377000));
```
Comment (JaySon-Huang) [inline dbms/src/Flash/tests/gtest_log_search.cpp:126]:
```suggestion
    ASSERT_TRUE(FilterFileByDatetime("/1/proxy.2021-10-09T14:50:55.123.log", {"/1/test-err.log"}, 1633855377000));
```
Should change the "proxy.log.xxx" -> "proxy.xxx.log"

**Review by JaySon-Huang — APPROVED** [review]
lgtm for the changes

**Comment by hehechen** [comment]
> I remember once that because of the log filename format, tiup clinic can NOT filter the log file when collecting within a specified time range. @hehechen Can you also ensure whether that problem still exists? Should the tiflash log filename format align with tikv's newest one?

No problem now.

I tried in my cluster:
`tiup diag collect ctl-tidb-test -f="-4h" -t="-2h"`
image

`tiup diag collect ctl-tidb-test -f="-400h" -t="-2h"`
image

**Pull request merged**